app/helpers/katello/katello_urls_helper.rb

```diff
@@ -1,15 +1,36 @@
 require 'uri'
+
 module Katello
   module KatelloUrlsHelper
+    def foreman_settings_url(schema = nil)
+      if schema
+        Settings[:foreman_url].sub(%r|^.*:|, "#{schema}:")
```

Comment:
If you define `foreman_settings_url(schema = 'http')` the method can be simplified to:

```ruby
def foreman_settings_url(schema = 'http')
  Settings[:foreman_url].sub(%r|^.*?:|, "#{schema}:")
end
```